### ISSUE TITLE
More in-depth texturing/variant tools

### ISSUE THREAD

**Issue by DankLondon** [TeamOpenIndustry/ImmersiveRailroading#1416]
**Describe the suggestion**  
More in-depth variant creation via JSON or CAML. 

**Applications of the suggestion**  
Being able to tell IR which textures exactly need to change per variant could allow for performance optimizations on packs with large amounts of liveries that utilize several texture sheets. For example, an engine which has everything below the frame on one sheet would only need that texture changed once or twice across a whole model (think weathered variants), but as it currently is implemented would require the same texture copied over and over again because it's different from the base texture. Also being able to manually override MTL filepaths through IR rather than manual MTL editing could allow for even more optimizations (parts like couplers which never change texture-wise), so modelers who aren't aware or confident in MTL editing can still save texturespace.

**Comment by Larky2k** [comment]
The current system you only need to put in the texture sheet that changes?

**Comment by SteampsonJake** [comment]
Yes, but if you have one of those sheets for liveries 1-3, and a different one for liveries 4-6, you need to copy it three times for those other liveries

**Comment by DankLondon** [comment]
> The current system you only need to put in the texture sheet that changes?

Aye but like Jake said, some liveries require certain sheets while most others might not. The default behavior could/should still be the way we know and have already, but this way would give modelers more control over what gets used and where. Reduces bloat and empowers creativity.

**Comment by PeterKitsune** [comment]
If you, for instance, have different liveries for the wagon being green, blue, and grey, and also different liveries for the wagon having coal, gravel, and sand in it, then you have 9 liveries (green coal, green gravel, green sand, blue coal, blue gravel, blue sand, grey coal, grey gravel, grey sand), but only six unique texture files.

The way we have it currently, the sheets on the default livery are the only ones we can skip in the liveries, meaning for the six unique texture files (three wagon colours and three loads), you need fourteen texture files (one of green, three each of blue and grey, one of coal, three each of gravel and sand). This also means that if you want to add another colour to the wagon, or another potential load to the wagon, you need to go through all the possible combinations to create a livery for it (both a folder and a json entry), which gets very tedious very quickly.

My suggestion would be to be able to have variant categories - e.g. the modeller creates two variant categories, one for colours and one for loads. The user can select one variant from each variant category. Different variant categories cannot overwrite the same file as each other.
